pioinstaller/core.py

```
@@ -31,22 +30,38 @@
 )
+
+_CACHED_RELEASE_URL = None
+
+
+def _get_release_url():
+    """Resolve latest release zip URL lazily with fallback and cache."""
+    # pylint: disable=protected-access
+    if hasattr(_get_release_url, "_cache"):
+        return _get_release_url._cache
+    try:
+        import requests  # noqa: WPS433
```

Ruff (0.12.2)

Unused `noqa` directive (unknown: `WPS433`)

Remove unused `noqa` directive

(RUF100)